src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's contact in an address book.
+ */
+public abstract class Contact {
+
+    public final String value;
+    private final boolean isPrivate;
+
+    public Contact(String value, boolean isPrivate) {
+        this.value = value.trim();
+        this.isPrivate = isPrivate;
+    }
+
+    @Override
```

Comment:
@xdshang To ensure that the method is overriding something, in this case the `toString`, `equals` and `hashCode` methods of the base `Object` class. AFAIK, all Java classes inherit from `Object`.